octodns/record/__init__.py

```diff
@@ -718,7 +718,10 @@ def process(self, value):
 
 
 class CnameValue(_TargetValue):
-    pass
+
+    @classmethod
+    def process(self, value):
+        return value.lower()
```

Comment:
I believe the `.lower()` will need to be up line ~717 in the parent class so that it'll fix ALIAS records as well.

Reply:
I'm not sure if you prefer to further revising other parts as simply mv it to `class _TargetValue(object)` will have some other issues:

```
======================================================================
ERROR: test_value_mixin_data (test_octodns_record.TestRecord)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pete/octodns/tests/test_octodns_record.py", line 152, in test_value_mixin_data
    'value': None
  File "/home/pete/octodns/octodns/record/__init__.py", line 365, in __init__
    self.value = self._value_type.process(data['value'])
  File "/home/pete/octodns/octodns/record/__init__.py", line 717, in process
    return value.lower()
AttributeError: 'NoneType' object has no attribute 'lower'
-------------------- >> begin captured logging << --------------------
Record: DEBUG: __init__: zone.name=unit.tests., type=AliasRecord, name=
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
```

Reply:
Or I can do a string detection in `class _TargetValue(object)` and decide to lower or not, what do you think?

Reply:
Yeah. I believe that's a case of where Alias is being used to test _TargetValue. Both CNAME and ALIAS could have a None value, just ALIAS has a test.

It should be fine to do an 

```python
if value:
   return value.lower()
```

There's no need to check for a string in that specific case as that'd go beyond the types of checking that octoDNS currently does (everywhere.)

Reply:
Fair enough, I also noticed PTR value also uses `_TargetValue`, added it here together(hope I'm not wrong 😓 )

Reply:
All updated and passed tests now 🎉